### DISCUSSION TITLE
Pass parameters to Plotter

### DISCUSSION THREAD

**Discussion by jonnymaserati** [marcomusy/vedo#560] [Ideas]
Hi there

Are you accepting pull requests?

I've made a couple of tweaks so that I can pass parameters to an instance of `vtkCubeAxesActor` in the `addons` module (for when the parameter `axes=8` is provided to `show`). This allows for the auto rotating labels to be used rather than the fixed ones when the custom axes are used.

I've set it to pass `kwargs` which are used to call methods on the object which means that all the methods from `vtk` can then be called via `vedo`.

https://user-images.githubusercontent.com/41046859/146171047-86bfbb9d-c0ff-4833-8b65-addf02e3b180.mp4

I need to play some more with label formatting etc, but seems to be working fine.

**Comment by marcomusy** [comment]
Hi Jonny

Yes - but I would rather be in favor of subclassing the `vtkCubeAxesActor` to something like:

```python
# in vedo.addons:
class CubeAxes(vtkCubeAxesActor):
   ...

# in main:
from vedo import *
plt = Plotter()
plt += Sphere()
plt += CubeAxes(...) # all options go here
plt.show()

# equivalent to:
from vedo import *
sph = Sphere()
ca = CubeAxes(sph, **kwargs)
show(sph, ca)
```

what do you think?

**Reply by jonnymaserati** [reply]
Indeed... I played some more and found it's actually not trivial to process the `kwargs` since `vtk` has some tricky indexing that is not simple to implement, so I think your suggestion is better.

In the meantime I have modified your code to allow me to pass my own axis instance when I use the `axes=8` parameter, which is also an option, but I didn't realize I could just initiate the `Plotter()` and add the axis to it... this is simpler/easier indeed!

Unfortunately, I'm struggling to change the axes label formatting for the `vtkCubeAxesActor` instance... I posted for some help [here](https://discourse.vtk.org/t/not-able-to-format-vtkcubeaxesactor-axis-labels/7411?u=jonnymaserati) but that forum doesn't seem to be that active? Do you have any suggestions/recommendations? The number format defaults to scientific notation and I can't change it to show the full number with no decimals... frustrating! I get the impression that some of the VTK Python wrappers have not been fully implemented or maintained?

**Reply by marcomusy** [reply]
> Indeed... I played some more and found it's actually not trivial to process the `kwargs` since `vtk` has some tricky indexing that is not simple to implement, so I think your suggestion is better.

OK _ if you want to give it a try you are welcome to !


> Unfortunately, I'm struggling to change the axes label formatting for the `vtkCubeAxesActor` instance... I posted for some help [here](https://discourse.vtk.org/t/not-able-to-format-vtkcubeaxesactor-axis-labels/7411?u=jonnymaserati) but that forum doesn't seem to be that active? Do you have any suggestions/recommendations? The number format defaults to scientific notation and I can't change it to show the full number with no decimals... frustrating! I get the impression that some of the VTK Python wrappers have not been fully implemented or maintained?

If they don't answer .. probably there is no solution.. I guess the `vtkCubeAxesActor` class is a tricky one and nobody is that eager to make fixes.
